includes/class-dispatcher.php

```diff
@@ -169,6 +160,76 @@ public static function send_to_followers( $outbox_item_id, $batch_size = 50, $of
 		}
 	}
 
+	/**
+	 * Retry sending to followers.
+	 *
+	 * @param array $inboxes        The inboxes.
+	 * @param int   $outbox_item_id The Outbox item ID.
+	 * @param int   $attempt        The attempt number.
+	 */
+	public static function retry_send_to_followers( $inboxes, $outbox_item_id, $attempt = 1 ) {
+		$retries = self::send_to_inboxes( $inboxes, $outbox_item_id );
+
+		// Retry failed inboxes.
+		if ( ++$attempt < 3 && ! empty( $retries ) ) {
+			self::schedule_retry( $retries, $outbox_item_id, $attempt );
+		}
+	}
+
+	/**
+	 * Send to inboxes.
+	 *
+	 * @param array $inboxes        The inboxes.
+	 * @param int   $outbox_item_id The Outbox item ID.
+	 * @return array The failed inboxes.
+	 */
+	private static function send_to_inboxes( $inboxes, $outbox_item_id ) {
+		$json    = Outbox::get_activity( $outbox_item_id )->to_json();
+		$actor   = Outbox::get_actor( \get_post( $outbox_item_id ) );
+		$retries = array();
+
+		foreach ( $inboxes as $inbox ) {
+			$result = safe_remote_post( $inbox, $json, $actor->get__id() );
+
+			if ( is_wp_error( $result ) ) {
+				$retries[] = $inbox;
```

Comment:
should we only retry certain http error codes?

Reply:
https://www.restapitutorial.com/advanced/responses/retries

Reply:
Fixed in [5585740](https://github.com/Automattic/wordpress-activitypub/pull/1336/commits/55857407aa55bb9b6ca3b377e3591ebabe69a221)